join.spec.md

```diff
@@ -228,22 +218,23 @@ The role of the router is to break apart an incoming query against the published
 
 A *realized selection set* is a selection within an inbound query, with particular type bounds, which will be issued against a particular subgraph. We denote the realized type of this selection set as `subgraph::(Type Bounds)`. Within this section, we use the special subgraph name `self` to refer to the exported composed schema.
```

Comment:
I don't really understand anything in this entire data model section.

I feel like probably the data model should come before the definition of the directives (and enum) to explain the overall concept of why this spec exists. But it also feels like the entire section needs to be rewritten from scratch.

Reply:
Fixed